core/shim/cmd/main.go

```diff
@@ -1,15 +1,23 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"io"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"os"
 	"os/exec"
+	"strings"
+	"time"
 )
 
 const (
-	stdinPath    = "/.dagger_meta_mount/stdin"
-	exitCodePath = "/.dagger_meta_mount/exitCode"
+	stdinPath     = "/.dagger_meta_mount/stdin"
+	exitCodePath  = "/.dagger_meta_mount/exitCode"
+	httpProxyHost = "localhost:14242"
```

Comment:
Until we build an engine image w/ CNI support all exec ops are in the same network namespace so I think this will create conflicts?

Fortunately it will soon be easy to include CNI support with the engine image!

Reply:
Right!

Do I try some random port with retry logic, or leave this feature be?

I'm trying to get python tests running in dagger but apparently there's no easy way for Python to communicate over unix sockets.

Currently for extensions, as far as I can see, it's `pythoruntime.go` that uses a `socat` to do the proxying.

Reply:
If you listen on `localhost:0` a random open port will be chosen for you, which you can then retrieve from the listener object with the right incantation (just learned this earlier today [here](https://github.com/dagger/dagger/pull/3647/files#diff-9d924a8c4bea3290367cd548454168f7ea6783a326a5fc5d0038a34dccf595c9R52-R61)).

So I think that'd work here; listen on random port, set env var based on that.

If someone is trying to run 10,000 shims on their box I guess port contention could matter, but I look forward to that bug report!

Reply:
Will do!

Also -- this is only enabled when using the (semi) internal nesting flag on exec.

Proxy kicks off only if DAGGER_HOST is set

Reply:
Done!